feat: PAYMCLOUD-395 Set minimum TLS version to 1.3 when possible via variables for consistency

ffppa opened the pull request:
### List of changes

Updated all configurations to use `minimum_tls_version` variables instead of hardcoding specific TLS versions. Default is set to TLS 1.3 where applicable. This change ensures flexibility, consistency, and ease of future updates across modules.

### Motivation and context

This change removes hardcoded TLS versions and replaces them with a variable, facilitating easier updates and maintaining consistency across modules. It ensures the default configuration is aligned with industry best practices by setting TLS 1.3 as the default version where possible.

### Type of changes

- [ ] Add new module
- [x] Update existing module
- [ ] Remove existing module

### Does this introduce a breaking change?

- [ ] Yes
- [x] No

### Other information

N/A

### Run checks

Useful commands to run checks on local machine

```sh
bash .utils/terraform_run_all.sh init local
pre-commit run -a
```

ffppa committed:
Set minimum TLS version via variables across modules

Updated configurations to reference `minimum_tls_version` variables instead of hardcoding TLS versions. This ensures consistency, enhances flexibility, and allows for easier updates across all modules. Default is set to TLS 1.3 where applicable.

pasqualedevita commented on app_gateway/main.tf:
```diff
@@ -282,7 +282,7 @@ resource "azurerm_application_gateway" "this" {
       "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
     ]
-    min_protocol_version = "TLSv1_2"
+    min_protocol_version = var.ssl_profiles[0].ssl_policy.min_protocol_version
```
app gateway with cipher suites TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 are equivalent to tls1.3.

Also change tls version on appgateway can create problems to clients

For these reasons better not change at the moment app gateway tls version.

all others are ok

ffppa replied:
Roll backed on min TLSv1_2 on this part

ffppa committed:
Merge branch 'main' into PAYMCLOUD-395-tls-upgrade-to-1.3

ffppa committed:
Set minimum SSL protocol version to TLSv1.2

Updated the SSL policy to explicitly set the minimum protocol version to TLSv1.2 instead of relying on a variable. This ensures consistent and secure configuration across environments.

ffppa committed:
Update default TLS version to TLSv1_2 in variables.tf

The default minimal_protocol_version has been changed from TLSv1_3 to TLSv1_2 to ensure better compatibility. This change aligns with broader support for clients that may not yet support TLSv1_3.
